spec/unit/facter/network_nexthop_ip_spec.rb

```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+describe Facter.fact(:network_nexthop_ip) do
+  before { Facter.clear }
+  after { Facter.clear }
+
+  describe 'network_nexthop_ip' do
+    context' returns gateway' do
+      before do
+        allow(Facter.fact(:kernel)).to receive(:value).and_return(:linux)
```

Comment:
Real facts return a string I think:
```suggestion
         allow(Facter.fact(:kernel)).to receive(:value).and_return('Linux') 
```